fuzz/src/process_onion_failure.rs

```diff
@@ -0,0 +1,170 @@
+// FUZZER COMMANDS:
+// RUSTFLAGS="--cfg hashes_fuzz --cfg=secp256k1_fuzz" cargo +nightly fuzz run --features "libfuzzer_fuzz" process_onion_failure_target
+// RUSTFLAGS="--cfg hashes_fuzz --cfg=secp256k1_fuzz" cargo +nightly fuzz coverage -v --features "libfuzzer_fuzz" process_onion_failure_target
+// llvm-cov show target/aarch64-apple-darwin/coverage/aarch64-apple-darwin/release/process_onion_failure_target --instr-profile=/Users/joost/repo/rust-lightning/fuzz/coverage/process_onion_failure_target/coverage.profdata --format=html --output-dir=reports
+
+use std::sync::Arc;
+
+use bitcoin::{
+	key::Secp256k1,
+	secp256k1::{PublicKey, SecretKey},
+};
+use lightning::{
+	blinded_path::BlindedHop,
+	ln::{
+		channelmanager::{HTLCSource, PaymentId},
+		msgs::OnionErrorPacket,
+	},
+	routing::router::{BlindedTail, Path, RouteHop, TrampolineHop},
+	types::features::{ChannelFeatures, NodeFeatures},
+	util::logger::Logger,
+};
+
+// Imports that need to be added manually
+use crate::utils::test_logger::{self};
+
+/// Actual fuzz test, method signature and name are fixed
+fn do_test<Out: test_logger::Output>(data: &[u8], out: Out) {
+	let mut read_pos = 0;
+	macro_rules! get_slice {
+		($len: expr) => {{
+			let slice_len = $len as usize;
+			if data.len() < read_pos + slice_len {
+				return;
+			}
+			read_pos += slice_len;
+			&data[read_pos - slice_len..read_pos]
+		}};
+	}
+
+	macro_rules! get_u8 {
+		() => {
+			get_slice!(1)[0]
+		};
+	}
+
+	macro_rules! get_u16 {
+		() => {
+			match get_slice!(2).try_into() {
+				Ok(val) => u16::from_be_bytes(val),
+				Err(_) => return,
+			}
+		};
+	}
+
+	macro_rules! get_u32 {
+		() => {
+			match get_slice!(4).try_into() {
+				Ok(val) => u32::from_be_bytes(val),
+				Err(_) => return,
+			}
+		};
+	}
+
+	macro_rules! get_u64 {
+		() => {
+			match get_slice!(8).try_into() {
+				Ok(val) => u64::from_be_bytes(val),
+				Err(_) => return,
+			}
+		};
+	}
+
+	macro_rules! get_bool {
+		() => {
+			get_slice!(1)[0] != 0
+		};
+	}
+
+	macro_rules! get_pubkey {
+		() => {
+			match PublicKey::from_slice(get_slice!(33)) {
+				Ok(val) => val,
+				Err(_) => return,
+			}
+		};
+	}
+
+	let secp_ctx = Secp256k1::new();
+	let logger: Arc<dyn Logger> = Arc::new(test_logger::TestLogger::new("".to_owned(), out));
+
+	let session_priv = match SecretKey::from_slice(get_slice!(32)) {
+		Ok(val) => val,
+		Err(_) => return,
+	};
+
+	let payment_id = match get_slice!(32).try_into() {
+		Ok(val) => PaymentId(val),
+		Err(_) => return,
+	};
+
+	let mut hops = Vec::<RouteHop>::new();
+	let hop_count = get_slice!(1)[0] as usize;
+	for _ in 0..hop_count {
+		hops.push(RouteHop {
+			pubkey: get_pubkey!(),
+			node_features: NodeFeatures::empty(),
+			short_channel_id: get_u64!(),
+			channel_features: ChannelFeatures::empty(),
```

Comment:
I didn't bother setting the features as they aren't used for the onion failure processing. Mainly because it seemed not straight-fwd to do it.